### PR TITLE
Add proxy support for the editor

### LINKED ISSUE #6100: Asset Library not working behind proxy (also applies to HTTPClient and HTTPRequest)
___

***Bugsquad edit:** No need to further confirm this issue. It's already been confirmed several times.*

___

**Operating system or device - Godot version:**
Ubuntu 16.04

**Issue description** (what happened, and what was expected):
When running godot, if we are behind a proxy, the environment http_proxy https_proxy not applied
**Steps to reproduce:**
Run godot behind a proxy

**Link to minimal example project** (optional but very welcome):

### PR DESCRIPTION
**Pull request by timothyqiu**
Follow-up of #47257.

* Adds proxy support for `HTTPRequest`.
* Adds `network/http_proxy/{host,port}` editor settings.
    * Labeled as "HTTP Proxy" and it's meant to be used for both HTTP and HTTPS requests. This is the same convention as seen in Android Studio's proxy settings.
* Makes Asset Library and Export Template Manager setup proxy according to the editor settings.

The proxy has to be set manually in Editor Settings. It would be a separate PR trying to figure out default proxy from environment variables on different platforms 😜

Fixes #6100.

### CODE REVIEW

**Comment by timothyqiu** [comment]
I can do a separate 3.x PR for these two proxy PRs after this one is merged. Adding the `cherrypick: 3.x` tag just as a reminder :smile:

**Comment by Calinou** [comment]
> The proxy has to be set manually in Editor Settings. It would be a separate PR trying to figure out default proxy from environment variables on different platforms stuck_out_tongue_winking_eye

On desktop platforms, I believe the `HTTP_PROXY` and `HTTPS_PROXY` environment variables are de facto standards nowadays.

**Comment by timothyqiu** [comment]
Yeah, these environment variables are definitely the way to go. Meanwhile, on Windows and macOS, desktop applications tend to honor the system proxy settings which have to be queried via API.

**Review by mhilbrunner — APPROVED** [review]

**Comment by mhilbrunner** [comment]
This is a straighforward change after the proxy PR. Looks good, and thanks for working on this!

**Comment by mhilbrunner** [comment]
I'll let @Faless merge.

**Comment by Faless** [comment]
Thanks!

**Comment by almusx** [comment]
Authenticated proxy doesnt work.

**Comment by Calinou** [comment]
> Authenticated proxy doesnt work.

Please open a new issue for this, while detailing your proxy configuration (which authentication method does it use?).

**Pull request merged**
